cinn/auto_schedule/task_registrar.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 CINN Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <gflags/gflags.h>
+
+#include <mutex>
+#include <string>
+
+#include "cinn/ir/ir_schedule.h"
+#include "cinn/optim/ir_copy.h"
+
+namespace cinn {
+
+namespace auto_schedule {
+
+// Global task registrar, used to save the initial ModuleExpr of each task.
+class TaskRegistrar {
```

Comment:
```suggestion
class TaskRegistry
```
registrar表示登记者，registry有注册库含义，更为合适